lib/classes/Utils.js

```diff
@@ -144,47 +145,111 @@ class Utils {
+    if (!this.fileExistsSync(trackingIdFilePath)) {
+      fs.writeFileSync(trackingIdFilePath, userId);
+    } else {
+      userId = fs.readFileSync(trackingIdFilePath).toString();
+    }
+
+    // function related information retrieval
+    const numberOfFunctions = _.size(serverless.service.functions);
+
+    const memorySizeAndTimeoutPerFunction = [];
+    if (numberOfFunctions) {
+      _.forEach(serverless.service.functions, (func) => {
+        const memorySize = Number(func.memorySize)
+          || Number(this.serverless.service.provider.memorySize)
+          || 1024;
+        const timeout = Number(func.timeout)
+          || Number(this.serverless.service.provider.timeout)
+          || 6;
+
+        const memorySizeAndTimeoutObject = {
+          memorySize,
+          timeout,
+        };
+
+        memorySizeAndTimeoutPerFunction.push(memorySizeAndTimeoutObject);
+      });
+    }
+
+    // event related information retrieval
+    const numberOfEventsPerType = [];
+    const eventNamesPerFunction = [];
+    if (numberOfFunctions) {
       _.forEach(serverless.service.functions, (func) => {
         if (func.events) {
+          const funcEventsArray = [];
+
           func.events.forEach((event) => {
             const name = Object.keys(event)[0];
-            const alreadyPresentEvent = _.find(events, { name });
+            funcEventsArray.push(name);
+
+            const alreadyPresentEvent = _.find(numberOfEventsPerType, { name });
             if (alreadyPresentEvent) {
               alreadyPresentEvent.count++;
             } else {
-              events.push({
+              numberOfEventsPerType.push({
                 name,
                 count: 1,
               });
             }
           });
+
+          eventNamesPerFunction.push(funcEventsArray);
         }
       });
     }
 
+    const hasCustomVariableSyntaxDefined = serverless.service
+      .defaults.variableSyntax !== '\\${([ :a-zA-Z0-9._,\\-\\/\\(\\)]+?)}';
+
     const auth = `${writeKey}:`;
 
     const data = {
-      userId: 'anonymousUser',
+      userId,
       event: 'Serverless framework usage',
       properties: {
-        commands: serverless.processedInput.commands,
-        operatingSystem: process.platform,
-        loadedPlugins,
-        serverlessVersion: serverless.version,
-        numberOfFunctions: _.size(serverless.service.functions),
-        events,
-        totalNumberOfEvents: events.length,
-        provider: serverless.service.provider,
-        isRunInsideService: (!!serverless.config.servicePath),
+        command: {
+          name: serverless.processedInput.commands.join(' '),
+          isRunInService: (!!serverless.config.servicePath),
+        },
+        service: {
+          numberOfCustomPlugins: _.size(serverless.service.plugins),
+          hasCustomResourcesDefined: (!!serverless.service.resources),
+          hasVariablesInCustomSectionDefined: (!!serverless.service.custom),
```

Comment:
This in `serverless.yml`:

```yml
custom:
```

will result in `null` for `serverless.service.custom` which in return will be translated to `false` (because of `!!`).